tests/test_aws.py

```diff
@@ -33,37 +48,88 @@ def test_session_env(monkeypatch):
     monkeypatch.undo()
 
 
-@pytest.mark.xfail(
-    (not(os.environ.get('GDALVERSION', '2.1').startswith('2.1')) or
-        'AWS_ACCESS_KEY_ID' not in os.environ or
-        'AWS_SECRET_ACCESS_KEY' not in os.environ),
-    reason="S3 raster access requires GDAL 2.1")
+@mingdalversion
+@credentials
```

Comment:
Multiple markers is much better. And `skipif` semantics are a better fit than `xfail`.